src/lib.ts

```diff
@@ -683,3 +687,80 @@ export function printEventLog(): void {
     );
   }
 }
+
+const rateCache = new Map<string, Map<Monster, number>>();
+
+export function barfEncounterRate(options: {
+  snapper?: boolean;
+  snapperPhylum?: Phylum;
+  olfact?: Monster;
+  longCon?: Monster;
+  motif?: Monster;
+  turtle?: Monster;
+  monkeyPoint?: Monster;
+  humanity?: boolean;
+}): Map<Monster, number> {
+  const olfact = options.olfact ?? get("olfactedMonster");
+  const longCon = options.longCon ?? get("longConMonster");
+  const motif = options.motif ?? get("motifMonster");
+  const turtle = options.turtle ?? get("_gallapagosMonster");
+  const monkeyPoint = options.monkeyPoint ?? get("monkeyPointMonster");
+  const snapper = options.snapper ?? myFamiliar() === $familiar`Red-Nosed Snapper`;
+  const snapperPhylum = options.snapperPhylum ?? Snapper.getTrackedPhylum();
+  const humanity = options.humanity ?? have($effect`Ew, The Humanity`);
+
+  const zoneMonsters = $monsters`garbage tourist, angry tourist, horrible tourist family`;
+  const encounterQueue = zoneMonsters.filter((m) =>
+    $location`Barf Mountain`.combatQueue.includes(`${m}`),
```

Comment:
I think I prefer `m.name` here but I think using the template literal below makes sense so maybe being consistent within this area is better